io/serialport/25-serial.js

```diff
@@ -27,99 +27,115 @@ module.exports = function(RED) {
         this.out = n.out || "char";
         this.waitfor = n.waitfor || "";
         this.responsetimeout = n.responsetimeout || 10000;
-    }
-    RED.nodes.registerType("serial-port",SerialPortNode);
+    
+        this.newPort = (serialPort) => {
+            serialPool.close(this.serialport,() => {});
```

Comment:
If the port has failed to open (if it's not there for example) - then will close work OK ? And if it is open then should we flush any unwritten data first before close ?